application/config/config.go

```diff
@@ -161,6 +161,7 @@ type Config struct {
 	runtimeName                  string
 	runtimeVersion               string
 	automaticScanning            bool
+	isSnykAutofixEnabled         concurrency.AtomicBool
```

Comment:
Most likely, we should have a capability to have scanners/infrastructure implementations provide their own configuration and merge it automatically to the top level.

Reply:
Yeah that sounds like a neat idea in a longer term indeed. I guess not only autofix is the candidate to go, but the whole bunch of settings e.g. from Code.

Reply:
That would be outside of the scope of this PR, right?

Reply:
My point is that it shouldn't be configurable in `config.go`, instead within Code scanner itself, if you want to have it under environment variable.

Reply:
Application config should be agnostic to scanner-level configuration, unless it's being exposed as a key to be configured on LS startup / or via user settings in IDE.

Reply:
Ok, I moved the settings into `code` package as a singletone too, and added tests. Commit: [fe494b0be564f88cbfc03954cac2c0ebf894df66 ](https://github.com/snyk/snyk-ls/pull/234/commits/fe494b0be564f88cbfc03954cac2c0ebf894df66)

Rejected options:
- moving config as a property of the scanner was less preferrable because the codeaction creation happens at the level of bundle, which has no reverse access to the scanner that created this bundle

Questions about tests:
- Currently, I kept the code for resetting settings defaults in `infrastructure/code` tests only in the respective places (where I test them(, but I would generally like some way to do resetting uniformly. I could import `code` from `testutils` and add `resetCodeSettings` in to the `t.Cleanup` in `testutils.UnitTest(t)` and Co. but that's subject to your decision -- because it makes top-level package aware of the bottom-level ones. Not sure you want that :D